### ISSUE TITLE
Ran into issue when running multiple containers

### ISSUE THREAD

**Issue by IxsharpxI** [lloesche/valheim-server-docker#267]
So i tried to run two of these containers and two worlds on the same machine. I made sure all the names were different and ports were not overlapping. on the second container i mapped them 2459-2561:2456-2458. I then setup a domain name and everything else mapped to the second server. However the first time i logged in i had to use the password of my original server and went to the new world. Then after that first time it was always the old password and took me to the original world. Its like i couldn't have two seperate accessible servers from the same location or perhaps the same steam account?

i also tried a completely separate vm to run the container in with the same ports and got the same result.

**Comment by IxsharpxI** [comment]
ok just to update. everything is working as expected normally. its only when externally accessing by domain name thats taking me back to my old server

**Comment by Toucan-Sam** [comment]
Are you specifying the port when you use your DNS to connect? server.example.com:2456
If so, then I suspect joining by DNS with a port in Valheim is broken. DNS support wasn't added until 0.147.3 on March 2nd.

**Comment by IxsharpxI** [comment]
No i was just using my domain name and separate subdomains. valheim.adfa.com and valheim2.adfa.com etc. And just letting nginx settings handle to port information.

**Comment by IxsharpxI** [comment]
Another update lol. So i can access it using the domainname:port. pretty strange problem and im not sure why its that way. for example.

using:
valheim.atfa.com
valheim2.atfa.com

always takes me to the original server

using:
valheim.atfa.com
valheim.atfa.com:2450

works as expected

thanks all

**Comment by Toucan-Sam** [comment]
It's important when you ask for help that you include a full description of your environment. Adding proxies introduces significantly different challenges. Not mad, just trying to help you ask better questions in the future.

So when you connect to valheim2.atfa.com it does a DNS lookup. That lookup resolves your IP address. If you haven't specified a port, Valheim will use the default port. valheim.atfa and valheim2.afta both have the same IP address.

DNS doesn't include a port. You can't say valheim2.atfa connects to this port. That's what an SRV record is for, but I don't think Valheim supports SRV records.

So in this case, when you don't specify a port, you end up at whatever server is running on the default one.

**TLDR?** No point in having multiple DNS entries (for Valheim) if you only have one IP address.

HTTP works differently because it supports a feature called SNI which specifically handles hostnames. So that might be why you thought it would work.

There are even more challenges when you add UDP to the mix which is the protocol that most games use to communicate.

**Comment by lloesche** [comment]
@IxsharpxI: what @Toucan-Sam wrote is spot on. A domain name points to an IP only, not a port (well an A, AAAA or CNAME record specifically). So as long as you only have one IP you'll always have to specify the ports you want to connect to. It's only when you have two IPs that you can create a setup where two domain names end up on two different server instances.

Does this resolve the issue for you?

**Comment by IxsharpxI** [comment]
yes thank you this is resolved
